### DISCUSSION TITLE
jsPsych data not appearing in Pavlovia

### DISCUSSION THREAD

**Discussion by ellamoeck** [jspsych/jsPsych#896] [Q&A] [answered]
Hi all,
I've successfully got my JsPsych experiment onto Pavlovia and it runs perfectly both on mine and another person's computer. However, I cannot get the data to save both in the pilotting and running modes. It seems like Pavlovia is not registering that the experiment is being run, because no credits are being deducted, nothing appears in the Sessions window, and I have no "data" folder being created in the repository. When I click "download data" a zip folder downloads, but I get an error when I try to open it (presumably because it's empty). 

I'm not sure if it's to do with an error in my jsPsych code (attached), or whether it's a Pavlovia issue. Does anyone have any insight or has encountered a similar issue before?
Thanks
Ella

[index.html.zip](https://github.com/jspsych/jsPsych/files/4951011/index.html.zip)

**Comment by vijaymarupudi** [comment]
This is more of a question to the Pavlovia (i.e. PsychoPy forums) I think. Nothing obvious jumps out to me.

Can you check for errors in the console or Network Monitor when the data is supposed to be posted? 

In pilot mode, I don't believe Pavlovia allows for saving of data (it's been a while)? In running mode it is supposed to.

console: 

Firefox: https://developer.mozilla.org/en-US/docs/Tools/Web_Console/Opening_the_Web_Console\
Chrome: https://developers.google.com/web/tools/chrome-devtools/open#console

network:

Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor
Chrome: https://developers.google.com/web/tools/chrome-devtools/network/

**Comment by ellamoeck** [comment]
Thanks Vijay. I'll check for errors in the console and network and post to the Pavlovia forums.

**Reply by jodeleeuw** [reply]
If you find a solution please update us!

**Comment by ellamoeck** [comment, marked as answer]
I did find a solution - a few people had posted similar issues on the Pavlovia PsychoPy discussion boards. 

Basically I had the connection to Pavlovia (starting and ending) in the overarching timeline, rather than the Experiment timeline (which is what I was cueing in the jspsych.init part of the code), so the two were not communicating properly. The other issue was that the end connection needed to occur before the end of the study. I have now placed it after the last bit of data is collected but before the debrief, so that there is sufficient time (while participants are reading the debrief) for the two to speak to each other and the data to save. Apparently you can expect lags of a few seconds for the data to save, so it's wise to have the pavlovia_finish variable _not_ be the last thing in the timeline.

Hopefully that all makes sense!

**Comment by Julie-DW** [comment]
Hi Ellamoeck, 

I permit myself to post a commentary in this discussion because I am a french psychology PhD student (University of Picardie Jules Verne), totaly novice with coding. In order to assess decision making skills, I have learn to code with JsPsych, and have created an experiment on JsPsych with videos (with Jspsych forum help). This experiment run efficiently locally on my computer.

To lead further research (especially due the sanitary situation, we would like to run this experiment on Pavlovia. For this, I try since weeks now, to follow these instructions : https://pavlovia.org/js-psych 

It seems to be not so hard, but as I am a novice, I think I do not do this properly. With your commentary, I see you successfully run your JsPsych experiment on Pavlovia, that is why I wonder if it would be possible to explain me how to do that too, or advice me a step by step tutorial ? Especially the part with GIT ? 

Thank you very much in advance for your answer, 
Best regards, 
Julie

**Reply by jodeleeuw** [reply]
Hi Julie, 

I don't have much experience working with Pavlovia, so hopefully @ellamoeck is able to offer some suggestions. If you are having a difficult time getting Pavlovia to work, you may find [cognition.run](https://www.cognition.run) to be simpler to use. It also has the benefit of being a free service.

Good luck!
Josh

**Reply by ellamoeck** [reply]
Hi Julie,
Happy to try and help as best I can, though I am not an expert by any means. Is it getting the git repository up and communicating with the version on your computer that you're having trouble with? Where have you gotten to so far?
Ella

**Reply by Julie-DW** [reply]
Hi Josh and Ella, 

First of all, thank you very much for your answers.

@jodeleeuw thank you very much for your advice about cognition.run, I didn't know this platform and, indeed, it is simpler to use, I successfully run my experiment on it (I just think I have to compress my videos because my stimuli do not upload for the moment, I think they are too heavy because when I try with another shorter video, it runs perfectly). 

I will talk about this platform to my PhD advisor, and while waiting for him to validate the use of this platform, I will continue to try to make it work on Pavlovia as well (just in case), for this, thank you in advance @ellamoeck . This is exactly my problem yes. I have downoaleded the git scm, and try to create a new repository with the codes indicated but I do not understand how to use the git scm on my computer (I used to work with Atom) and where should I enter the code mentionned, differently to the git Lab which appear more understandable to me. 

I do not know if I am clear and if I correctly follow the steps. In any case, thank you in advance for any advice you can give me. 
Have a good day and thanks again, 
Julie

**Reply by Julie-DW** [reply]
Hi @jodeleeuw, 

I permit myself to send you another message about cognition.run : is this platform capable of support videos stimuli (less than 5min for for each of the 3 videos) ? I have compressed my videos stimuli in order to they are less heavy to load, but it seems still hard to load. 

Thank you very much in advance, 
Best regards, 
Julie
